diff --git a/lib/config.js b/lib/config.js
index 22ef6d21b76db22df630996387c3d0cd8a302b3d..37685a36a919b20bb7ce45604de753f4c922c604 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -69,6 +69,11 @@ export function getTargetParts(target) {
             arch = "arm64";
             abi = "gnu";
             break;
+        case "aarch64-unknown-linux-musl":
+            platform = "linux";
+            arch = "arm64";
+            abi = "musl";
+            break;
         case "x86_64-apple-darwin":
             platform = "darwin";
             arch = "x64";
diff --git a/lib/lib.d.ts b/lib/lib.d.ts
index 5596ed0016aecf4a738f94852f9786b3c68db479..c3a55013d03372cf1fe2d29ebbc732cd8374d36c 100644
--- a/lib/lib.d.ts
+++ b/lib/lib.d.ts
@@ -1,4 +1,4 @@
-export declare const TARGETS: readonly ["aarch64-apple-darwin", "aarch64-unknown-linux-gnu", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu", "x86_64-unknown-linux-musl", "x86_64-pc-windows-msvc"];
+export declare const TARGETS: readonly ["aarch64-apple-darwin", "aarch64-unknown-linux-gnu", "aarch64-unknown-linux-musl", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu", "x86_64-unknown-linux-musl", "x86_64-pc-windows-msvc"];
 export type Target = (typeof TARGETS)[number];
 export declare enum Optimize {
     Debug = "Debug",
diff --git a/lib/lib.js b/lib/lib.js
index 3fe13998910a3ad4e8511871d803de7b473c5cd1..4c41b9b282b1e8728aaba0a61b2eb205d9d07cad 100644
--- a/lib/lib.js
+++ b/lib/lib.js
@@ -7,6 +7,7 @@ const require = createRequire(import.meta.url);
 export const TARGETS = [
     "aarch64-apple-darwin",
     "aarch64-unknown-linux-gnu",
+    "aarch64-unknown-linux-musl",
     "x86_64-apple-darwin",
     "x86_64-unknown-linux-gnu",
     "x86_64-unknown-linux-musl",
@@ -103,6 +104,8 @@ export function getZigTriple(target) {
             return "x86_64-linux-musl";
         case "aarch64-unknown-linux-gnu":
             return "aarch64-linux-gnu";
+        case "aarch64-unknown-linux-musl":
+            return "aarch64-linux-musl";
         case "x86_64-apple-darwin":
             return "x86_64-macos-none";
         case "aarch64-apple-darwin":
@@ -164,7 +167,7 @@ export function getTarget(platform, arch) {
     }
     else if (platform === "linux") {
         const abi = isMusl() ? "musl" : "gnu";
-        if (arch === "arm64" && abi === "gnu") {
+        if (arch === "arm64") {
             return `aarch64-unknown-linux-${abi}`;
         }
         if (arch === "x64") {
diff --git a/ts/config.ts b/ts/config.ts
index 11c124021a108c030d3cb1c7022484d670c45b11..dfc29ae9033987b305fda806db1811c691ec3eaa 100644
--- a/ts/config.ts
+++ b/ts/config.ts
@@ -100,6 +100,11 @@ export function getTargetParts(target: Target): {platform: NodeJS.Platform; arch
       arch = "arm64";
       abi = "gnu";
       break;
+    case "aarch64-unknown-linux-musl":
+      platform = "linux";
+      arch = "arm64";
+      abi = "musl";
+      break;
     case "x86_64-apple-darwin":
       platform = "darwin";
       arch = "x64";
diff --git a/ts/lib.ts b/ts/lib.ts
index 940845e55a28b9742005423e4c34a905c9447937..cc72bfa1f51c160662c5c2aa47c1ea7601cf48e4 100644
--- a/ts/lib.ts
+++ b/ts/lib.ts
@@ -9,6 +9,7 @@ const require = createRequire(import.meta.url);
 export const TARGETS = [
   "aarch64-apple-darwin",
   "aarch64-unknown-linux-gnu",
+  "aarch64-unknown-linux-musl",
   "x86_64-apple-darwin",
   "x86_64-unknown-linux-gnu",
   "x86_64-unknown-linux-musl",
@@ -121,6 +122,8 @@ export function getZigTriple(target: Target): string {
       return "x86_64-linux-musl";
     case "aarch64-unknown-linux-gnu":
       return "aarch64-linux-gnu";
+    case "aarch64-unknown-linux-musl":
+      return "aarch64-linux-musl";
     case "x86_64-apple-darwin":
       return "x86_64-macos-none";
     case "aarch64-apple-darwin":
@@ -189,7 +192,7 @@ export function getTarget(platform: NodeJS.Platform, arch: NodeJS.Architecture):
     }
   } else if (platform === "linux") {
     const abi = isMusl() ? "musl" : "gnu";
-    if (arch === "arm64" && abi === "gnu") {
+    if (arch === "arm64") {
       return `aarch64-unknown-linux-${abi}`;
     }
     if (arch === "x64") {
